cmd/shell.go

```diff
@@ -435,6 +437,17 @@ func ReadConfigFile(filename string, out interface{}) error {
 	return decodeJSONStrict(file, out)
 }
 
+// ValidatorForConfigDuration enables registration of our custom config.Duration
+// type as a time.Duration and performing validation on the configured value
+// using the standard suite of validation functions.
+func ValidatorForConfigDuration(field reflect.Value) interface{} {
+	if c, ok := field.Interface().(config.Duration); ok {
+		return c.Duration
+	}
+
+	return nil
```

Comment:
I think we actually want to return `reflect.Invalid` in this case.

Returning `nil` causes us to hit [this case](https://github.com/letsencrypt/validator/blob/2ed8f7d619aca77c29bca2a743425a322e2dd789/util.go#L32-L34), which in turn gets caught by [this case](https://github.com/letsencrypt/validator/blob/2ed8f7d619aca77c29bca2a743425a322e2dd789/validator.go#L103), which has the possibility of just blithely continuing forward.

Returning `reflect.Invalid` instead causes us to hit [this case](https://github.com/letsencrypt/validator/blob/2ed8f7d619aca77c29bca2a743425a322e2dd789/util.go#L39-L40), which at first gets handled by the [same case](https://github.com/letsencrypt/validator/blob/2ed8f7d619aca77c29bca2a743425a322e2dd789/validator.go#L103) in `traverseField`, but actually ends up falling into [this conditional](https://github.com/letsencrypt/validator/blob/2ed8f7d619aca77c29bca2a743425a322e2dd789/validator.go#L114) which _always_ returns an error. Returning an error is exactly what we want, if we were expecting a config.Duration but somehow magically didn't get one.

Reply:
![image](https://github.com/letsencrypt/boulder/assets/2382565/7764ec15-dba2-404d-a82c-1cbe7a6ba2a0)